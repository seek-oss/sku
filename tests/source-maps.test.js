const path = require('path');

const {
  runSkuScriptInDir,
  dirContentsToObject,
} = require('@sku-private/test-utils');

const appDir = path.dirname(
  require.resolve('@sku-fixtures/source-maps/sku.config.js'),
);

describe('source-maps', () => {
  describe('build', () => {
    beforeAll(async () => {
      await runSkuScriptInDir('build', appDir);
    });

    it('should generate the expected files', async () => {
      const files = await dirContentsToObject(`${appDir}/dist`);
      expect(files).toMatchSnapshot();
    });
  });
});
