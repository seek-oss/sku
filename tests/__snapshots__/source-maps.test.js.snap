// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`source-maps > build > should generate the expected files 1`] = `
{
  "2.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "2.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "index.html": "<!DOCTYPE html>
<html>
  <head>
    <meta charset="UTF-8" />
    <title>hello-world</title>
    <meta name="viewport" content="width=device-width, initial-scale=1" />
    
<link data-chunk="main" rel="preload" as="script" href="/static/source-maps/runtime.js">
<link data-chunk="main" rel="preload" as="script" href="/static/source-maps/2.js">
<link data-chunk="main" rel="preload" as="script" href="/static/source-maps/main.js">
  </head>
  <body>
    <div id="app"><div>Hello source maps</div></div>
    <script id="__LOADABLE_REQUIRED_CHUNKS__" type="application/json">[]</script><script id="__LOADABLE_REQUIRED_CHUNKS___ext" type="application/json">{"namedChunks":[]}</script>
<script async data-chunk="main" src="/static/source-maps/runtime.js"></script>
<script async data-chunk="main" src="/static/source-maps/2.js"></script>
<script async data-chunk="main" src="/static/source-maps/main.js"></script>
  </body>
</html>",
  "main.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "main.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "runtime.js.map": "CONTENTS IGNORED IN SNAPSHOT TEST",
}
`;
